xdsl/dialects/experimental/air.py

```diff
@@ -0,0 +1,810 @@
+        dst_sizes: list[Operation | SSAValue] = []
+        while not parser.parse_optional_characters("]"):
+            dst_sizes.append(parser.parse_operand())
+            parser.parse_optional_characters(",")
+        parser.parse_characters("[")
+        dst_strides: list[Operation | SSAValue] = []
+        while not parser.parse_optional_characters("]"):
+            dst_strides.append(parser.parse_operand())
+            parser.parse_optional_characters(",")
+        parser.parse_characters(",")
+        src = parser.parse_operand()
+        parser.parse_characters("[")
+        src_offsets: list[Operation | SSAValue] = []
+        while not parser.parse_optional_characters("]"):
+            src_offsets.append(parser.parse_operand())
+            parser.parse_optional_characters(",")
+        parser.parse_characters("[")
+        src_sizes: list[Operation | SSAValue] = []
+        while not parser.parse_optional_characters("]"):
+            src_sizes.append(parser.parse_operand())
+            parser.parse_optional_characters(",")
+        parser.parse_characters("[")
+        src_strides: list[Operation | SSAValue] = []
+        while not parser.parse_optional_characters("]"):
+            src_strides.append(parser.parse_operand())
+            parser.parse_optional_characters(",")
+        parser.parse_characters(")")
+        parser.parse_optional_attr_dict()
+        parser.parse_characters(":")
+        parser.parse_characters("(")
+        parser.parse_type()
+        parser.parse_characters(",")
+        parser.parse_type()
+        parser.parse_characters(")")
+
+        return DmaMemcpyNdOp(
+            None,
+            dst,
+            dst_offsets,
+            dst_sizes,
+            dst_strides,
+            src,
+            src_offsets,
+            src_sizes,
+            src_strides,
+        )
+
+
+@irdl_op_definition
+class ExecuteOp(IRDLOperation):
+    name = "air.execute"
+
+    async_dependencies = var_operand_def(AsyncTokenAttr)
+    async_token = result_def(AsyncTokenAttr)
+    results_ = var_result_def(Attribute)
+    body = region_def()
+
+    traits = traits_def(
+        lambda: frozenset([SingleBlockImplicitTerminator(ExecuteTerminatorOp)])
+    )  # TODO: solve dependency
+
+    def __init__(
+        self,
+        async_dependencies: list[Operation | SSAValue],
+        result_types: list[Attribute] | None,
+        body: Region,
+    ):
+        super().__init__(
+            operands=[async_dependencies],
+            result_types=[AsyncTokenAttr(), result_types],
+            regions=[body],
+        )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> ExecuteOp:
+        async_dependencies: list[Operation | SSAValue] = []
+        if parser.parse_optional_characters("["):
+            while not parser.parse_optional_characters("]"):
+                async_dependencies.append(parser.parse_operand())
+                parser.parse_optional_characters(",")
```

Comment:
By doing this `async_dependencies` would be of type `list[SSAValue] | None` instead of `list[Operation | SSAValue]`. Is it fine to drop `Operation` from the type like this?